cirq-core/cirq/ops/cxswap_czswap.py

```diff
@@ -0,0 +1,146 @@
+# Copyright 2024 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Optional, Tuple
+
+import numpy as np
+
+import cirq
+from cirq.ops import gate_features, raw_types
+
+
+class CZSWAPGate(gate_features.InterchangeableQubitsGate, raw_types.Gate):
+    r"""CZSWAP gate.
+
+    This gate combines the CZ and SWAP gates.
+
+    The unitary matrix of this gate is:
+
+    $$
+    \begin{bmatrix}
+        1 & 0 & 0 & 0 \\
+        0 & 0 & 1 & 0 \\
+        0 & 1 & 0 & 0 \\
+        0 & 0 & 0 & -1
+    \end{bmatrix}
+    $$
+
+    Up to a global phase, CZSWAP(0,1) is equivalent to applying the following:
+    H(0)
+    CX(0,1)
+    CX(1,0)
+    H(1)
+
+    """
+
+    def __init__(self) -> None:
+        super(CZSWAPGate, self)
+
+    def _num_qubits_(self) -> int:
+        return 2
+
+    def _unitary_(self) -> Optional[np.ndarray]:
+
+        # fmt: off
+        return np.array(
+            [
+                [1, 0, 0, 0],
+                [0, 0, 1, 0],
+                [0, 1, 0, 0],
+                [0, 0, 0, -1],
+            ]
+        )
+        # fmt: on
+
+    def _has_unitary_(self):
+        return True
+
+    def _decompose_(self, qubits):
+        a, b = qubits
+        yield cirq.H(a)
```

Comment:
I think this should reflect the name of the gate (so CX followed by SWAP)

Reply:
Good point, changed!